utils/src/main/scala/za/co/absa/hermes/utils/HelperFunctions.scala

```diff
@@ -53,9 +55,11 @@ object HelperFunctions {
       * @param arrayType ArrayType
       */
     def flattenStructArray(path: String, fieldNamePrefix: String, structField: StructField, arrayType: ArrayType): Unit = {
-      val maxInd = df.agg(max(expr(s"size($path${structField.name})"))).collect()(0)(0).toString.toInt
+      val maxLengths = (Seq(df) ++ Seq(dfsForArraysLength: _*))
+        .map(_.agg(max(expr(s"size($path${structField.name})"))).collect()(0)(0).toString.toInt)
```

Comment:
This is little hard to understand. And why the `.toString.toInt`? Wouldn't `.toInt` be enough?
```suggestion
      val firstRow = 0
      val firstColumn = 0
      val maxLengths = (Seq(df) ++ Seq(dfsForArraysLength: _*))
        .map(_.agg(max(expr(s"size($path${structField.name})"))).collect()(firstRow)(firstColumn).toInt)
```

Reply:
Please let me know if it is easier to understand now, I changed it a little. Can't be just `toInt` as result of `[...](0)(0)` is `Any`, but I cut the expression on `Row` level and used `getInt` instead.